easybuild/easyblocks/a/amber.py

```diff
@@ -0,0 +1,187 @@
+
+@author: Benjamin Roberts (The University of Auckland)
+@author: Kenneth Hoste (Ghent University)
+"""
+import os
+
+from easybuild.easyblocks.generic.configuremake import ConfigureMake
+from easybuild.framework.easyconfig import CUSTOM, MANDATORY, BUILD
+import easybuild.tools.environment as env
+from easybuild.tools.run import run_cmd
+from easybuild.tools.modules import get_software_root
+import easybuild.tools.toolchain as toolchain
+
+class EB_Amber(ConfigureMake):
+    """Easyblock for building and installing Amber"""
+
+    @staticmethod
+    def extra_options(extra_vars=None):
+        """Extra easyconfig parameters specific to ConfigureMake."""
+        extra_vars = dict(ConfigureMake.extra_options(extra_vars))
+        extra_vars.update({
+            # 'Amber': [True, "Build Amber in addition to AmberTools", CUSTOM],
+            'patchlevels': ["latest", "(AmberTools, Amber) updates to be applied", CUSTOM],
+        })
+        return ConfigureMake.extra_options(extra_vars)
+
+    def __init__(self, *args, **kwargs):
+        super(EB_Amber, self).__init__(*args, **kwargs)
+        self.already_extracted = False
+        self.build_in_installdir = True
+
+    def extract_step(self):
+        """Only extract from the tarball if this has not already been done."""
+        if (self.already_extracted == True):
+            pass
+        else:
+            self.cfg['unpack_options'] = "--strip-components=1"
+            super(EB_Amber, self).extract_step()
+            self.already_extracted = True
+
+    def patch_step(self, **kw):
+        env.setvar('AMBERHOME', self.installdir)
+        if self.cfg['patchlevels'] == "latest":
+            cmd = "./update_amber --update"
+            # This needs to be run multiple times, more's the pity.
+            run_cmd(cmd, log_all=True)
+            run_cmd(cmd, log_all=True)
+        else:
+            for (tree, patch_level) in zip(['AmberTools', 'Amber'], self.cfg['patchlevels']):
+                if patch_level == 0: continue
+                cmd = "./update_amber --update-to %s/%s" % (tree, patch_level)
+                # This needs to be run multiple times, more's the pity.
+                run_cmd(cmd, log_all=True)
+                run_cmd(cmd, log_all=True)
+        return super(EB_Amber, self).patch_step(**kw)
+
+    def configure_step(self):
+        cmd = "%(preconfigopts)s ./configure %(configopts)s" % {
+            'preconfigopts': self.cfg['preconfigopts'],
+            'configopts': self.cfg['configopts']
+        }
+        (out, _) = run_cmd(cmd, log_all=True, simple=False)
+        
+        return out
+
+    def build_step(self):
+
+        # Set the AMBERHOME environment variable
+        env.setvar('AMBERHOME', self.installdir)
+        try:
+            os.chdir(self.installdir)
+        except OSError, err:
+            self.log.error("Could not chdir to %(amberhome)s: %(error)s" % { 'amberhome': self.installdir, 'error': err })
```

Comment:
This is the problem. We don't use this anymore. Replace it by:
```python
raise EasyBuildError("Could not chdir to %s: %s",  self.installdir, err ) 
```
And also import it:
```python
from easybuild.tools.build_log import EasyBuildError
```